src/Scan.test.js

```diff
@@ -1352,4 +1357,62 @@ describe('Scan', () => {
       spy.mockRestore();
     });
   });
+
+  describe('Scan - renderActionChoiceModal', () => {
+    const args = [{ barcode: '12345', title: 'Book Title', materialType: { name: 'Book' } }, 'Circ Desk A'];
+
+    it('checkin method invokes action-choice modal', async () => {
+      const comp = new RawScan(basicProps);
+      comp.state.checkedinItem = { id: 123 };
+      comp.checkInData = {
+        item: {},
+      };
+      const setStateSpy = jest.spyOn(comp, 'setState');
+      await act(async () => comp.checkIn(CHECKIN_ACTIONS.ASK));
+      expect(setStateSpy).toHaveBeenCalledWith({ showActionChoiceModal: true });
+    });
```

Comment:
Spying on the internal state of a component is extremely brittle. If you want to test that Scan shows the action-choice modal, then mock the response that comes from `getLoanForItem()`, render Scan, and then use the UI to take the actions that would lead to showing the action-choice modal: enter a barcode, click the `Enter` button, and assert on the presence of the modal in the DOM.